tests/basic/execution.rs

```diff
@@ -169,3 +160,116 @@ fn max_steps_early_exit_scheduler() {
 
     assert_eq!(counter_clone.load(Ordering::SeqCst), 50 * 10);
 }
+
+#[test]
+#[should_panic]
+fn context_switches_outside_execution() {
+    shuttle::context_switches();
+}
+
+#[test]
+fn context_switches_atomic() {
+    check_random(
```

Comment:
Would be nice to do a slightly smaller version of this test (spawn 2 threads instead of 3?) and use `check_dfs`.

Reply:
Sounds good.